frontend/src/metabase/query_builder/components/template_tags/SnippetModal.jsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+
+import { t } from "ttag";
+
+import colors from "metabase/lib/colors";
+import Modal from "metabase/components/Modal";
+import Link from "metabase/components/Link";
+import Snippets from "metabase/entities/snippets";
+
+export default class SnippetSidebar extends React.Component {
+  componentDidMount() {
```

Comment:
I wonder if we should just add a `escapeKeyClosesModal` property to `Modal` and `Popover` or something? We'd also be able to check that it was the top most `Modal`/`Popover`. But this is fine for now.

Reply:
Turns out that was already the default behavior. I just needed to pass `onClose` to `Modal`.